doc/glossary.rst

```diff
@@ -1848,26 +1848,72 @@ See concept :term:`sample property`.
         See :ref:`group_cv`.
 
     ``sample_weight``
-        A relative weight for each sample.  Intuitively, if all weights are
-        integers, a weighted model or score should be equivalent to that
-        calculated when repeating the sample the number of times specified in
-        the weight.  Weights may be specified as floats, so that sample weights
-        are usually equivalent up to a constant positive scaling factor.
-
-        FIXME  Is this interpretation always the case in practice? We have no
-        common tests.
-
-        Some estimators, such as decision trees, support negative weights.
-        FIXME: This feature or its absence may not be tested or documented in
-        many estimators.
-
-        This is not entirely the case where other parameters of the model
-        consider the number of samples in a region, as with ``min_samples`` in
-        :class:`cluster.DBSCAN`.  In this case, a count of samples becomes
-        to a sum of their weights.
-
-        In classification, sample weights can also be specified as a function
-        of class with the :term:`class_weight` estimator :term:`parameter`.
+        A relative weight for each sample. Intuitively, if all weights are
+        integers, using them in a model or scorer is like duplicating each
+        sample as many times as the weight value. Weights may be specified as
+        floats, so that sample weights are usually equivalent up to a constant
+        positive scaling factor.
```

Comment:
equivalent to what? I don't understand this sentence.

Reply:
Invariance to the scale of weights is actually not guaranteed for all estimators or hyperparameter choices. So let's be more generic.

```suggestion
        floats to express fractional relative importance of data points with respect
        to one another others.
```

Reply:
I did not want to explicitly state that because we are not yet sure to which extent this is true or not: we do haven't started testing this aspect systematically, but we know that this is not the case for many important ones such as `LogisticRegression` and `Ridge`. Both those estimators have a regularization parameter whose effect depends on the scale of the weights.